test/common/quic/envoy_quic_client_session_test.cc

```diff
@@ -472,5 +474,48 @@ TEST_P(EnvoyQuicClientSessionTest, StatelessResetOnProbingSocket) {
   EXPECT_EQ(self_addr_->asString(), quic_connection_->self_address().ToString());
 }
 
+class MockOsSysCallsImpl : public Api::OsSysCallsImpl {
+public:
+  MOCK_METHOD(Api::SysCallSizeResult, recvmsg, (os_fd_t socket, msghdr* msg, int flags),
+              (override));
+  MOCK_METHOD(Api::SysCallIntResult, recvmmsg,
+              (os_fd_t socket, struct mmsghdr* msgvec, unsigned int vlen, int flags,
+               struct timespec* timeout),
+              (override));
+};
```

Comment:
We already have MockOsSysCalls does this in test/mocks/api/mocks.h

Reply:
but that mocks sendmsg and setting socket options too, which I don't want (b/c the peer_socket_ sends a msg to trigger EnvoyQuicClientConnection::onFileEvent)